src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> userCache = new HashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> {
+            if (userCache.containsKey(userId)) {
+                return userCache.get(userId);
+            } else {
+                try {
+                    Thread.sleep(1000);
```

Comment:
Using `Thread.sleep(1000);` in an asynchronous method can block the thread and is generally not recommended. Consider using a non-blocking delay mechanism or a different approach to simulate the delay.